src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    if (!this.hidden) {
+      this.hidden = true;
+    } else {
+      this.hidden = false;
+    }
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(creature) {
+    if (creature instanceof Carnivore) {
+      return;
+    }
+
+    if (!creature.hidden) {
+      creature.health -= 50;
+    }
+
+    if (creature.health <= 0) {
+      const index = Animal.alive.indexOf(creature);
+
+      Animal.alive.splice(index, 1);
```

Comment:
Don't manipulate with array in this way, use filter method instead

Reply:
why is it a bad idea?